web/blueprints/user/__init__.py

```diff
@@ -142,15 +145,119 @@ def static_datasheet(user_id):
+        applied_filter = True
+        interface = Interface.q.filter_by(mac=mac).first()
+
+        if interface is not None:
+            result = result.filter_by(id=interface.host.owner_id)
+        else:
+            result = result.filter_by(id=-1)
+    if ip_address:
+        applied_filter = True
+        ip = IP.q.filter_by(address=ip_address).first()
+
+        # result = result.filter(User.id.in_(User.q.join(Host).filter(
+        # Host.ips.has(IP.address.any_(IP.address == ip.address)))))
+
+        if ip is not None:
+            user_ids = list()
+
+            for host in ip.host:
+                user_ids.append(host.owner_id)
+
+            result = result.filter(User.id.in_(user_ids))
+        else:
+            result = result.filter_by(id=-1)
+    if trafficgroup_id and represents_int(trafficgroup_id):
+        applied_filter = True
+        memberships = Membership.q.join(TrafficGroup) \
+            .filter(TrafficGroup.id == trafficgroup_id) \
+            .filter(or_(
+            Membership.ends_at.is_(None),
+            Membership.ends_at > datetime.today()
+        )) \
+            .filter(or_(
+            Membership.begins_at.is_(None),
+            Membership.begins_at < datetime.today()
+        )) \
+            .all()
+
+        user_ids = list()
+
+        for membership in memberships:
+            user_ids.append(membership.user_id)
+
+        result = result.filter(User.id.in_(user_ids))
+    if propertygroup_id and represents_int(propertygroup_id):
+        applied_filter = True
+        memberships = Membership.q.join(PropertyGroup) \
+            .filter(PropertyGroup.id == propertygroup_id) \
+            .filter(or_(
+            Membership.ends_at.is_(None),
+            Membership.ends_at > datetime.today()
+        )) \
+            .filter(or_(
+            Membership.begins_at.is_(None),
+            Membership.begins_at < datetime.today()
+        )) \
+            .all()
+
+        user_ids = list()
+
+        for membership in memberships:
+            user_ids.append(membership.user_id)
+
+        result = result.filter(User.id.in_(user_ids))
+    if building_id and represents_int(building_id):
+        applied_filter = True
+
+        rooms = Room.q.filter_by(building_id=building_id)
+
+        room_ids = list()
+
+        for room in rooms:
+            room_ids.append(room.id)
+
+        result = result.filter(User.room_id.in_(room_ids))
+    if query:
+        applied_filter = True
+        result = result.filter(or_(
+            func.lower(User.name).like(
+                func.lower(u"%{0}%".format(query))),
+            func.lower(User.login).like(
+                func.lower(u"%{0}%".format(query))),
+            cast(User.id, Text).like(u"{0}%".format(query))
+        ))
+
+    return jsonify(items=[{
+        'id': found_user.id,
+        'name': found_user.name,
+        'login': found_user.login,
+        'room_id': found_user.room_id if found_user.room_id is not None else None
```

Comment:
This was my attempt in the first place but I then decided to seperate backend (search) from frontend (search results). But I can change this if wanted.

Reply:
Adding URLs to the search results doesn't break separation between backend and frontend. On the contrary, it improves separation, because the frontend no longer needs to know how backend URLs for users are structured.